tests/TenantDatabaseManagerTest.php

```diff
@@ -222,6 +223,159 @@
 
     $manager->setConnection('mysql2');
     expect($manager->databaseExists($name))->toBeTrue();
+
+    $manager->deleteDatabase($tenant);
+    expect($manager->databaseExists($name))->toBeFalse();
+});
+
+test('tenant database user can be created when using the permission controller MySQL manager', function () {
+    config([
+        'tenancy.database.managers.mysql' => PermissionControlledMySQLDatabaseManager::class,
+    ]);
+
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    $name = 'foo' . Str::random(8);
+
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+    ]);
+    $dbUser = $tenant->tenancy_db_username;
+
+    /** @var PermissionControlledMySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+    $manager->setConnection('mysql');
+
+    expect($manager->userExists($dbUser))->toBeTrue();
+});
+
+test('tenant database can be created on template tenant connection', function () {
+    config([
+        'tenancy.database.managers.mysql' => MySQLDatabaseManager::class,
+        'tenancy.database.template_tenant_connection' => 'mysql2',
+        'database.connections.mysql2' => [
+            'driver' => 'mysql',
+            'host' => 'mysql2',
+            'port' => 3306,
+            'database' => 'main',
+            'username' => 'root',
+            'password' => 'password',
+            'unix_socket' => env('DB_SOCKET', ''),
+            'charset' => 'utf8mb4',
+            'collation' => 'utf8mb4_unicode_ci',
+            'prefix' => '',
+            'prefix_indexes' => true,
+            'strict' => true,
+            'engine' => null,
+            'options' => extension_loaded('pdo_mysql') ? array_filter([
+                PDO::MYSQL_ATTR_SSL_CA => env('MYSQL_ATTR_SSL_CA'),
+            ]) : [],
+        ],
+    ]);
+
+    Event::listen(TenantCreated::class, JobPipeline::make([CreateDatabase::class])->send(function (TenantCreated $event) {
+        return $event->tenant;
+    })->toListener());
+
+    $name = 'foo' . Str::random(8);
+    $tenant = Tenant::create([
+        'tenancy_db_name' => $name,
+    ]);
+
+    /** @var MySQLDatabaseManager $manager */
+    $manager = $tenant->database()->manager();
+
+    // central was default connection so we are making sure
+    // that database did not create on central
+    $manager->setConnection('central');
+    expect($manager->databaseExists($name))->toBeFalse();
+
+    $manager->setConnection('mysql2');
+    expect($manager->databaseExists($name))->toBeTrue();
+});
+
+test('tenant database can be created on a foreign server by using the host from tenant config', function () {
```

Comment:
What's the point of this test as well? It's not very clear to me what we're trying to assert here.

Reply:
And the next test as well.

Reply:
> What's the point of this test as well?

This test ensures that tenant database connection can be established successfully when tenantConfig has a different host. For example, general host is `mysql`, but the developer wants to use a different host for different tenants. So he provides the host in tenantConfig. 
```php
Tenant::create([
        'tenancy_db_host' => 'mysql2',
    ]);
```
So the test ensures the connection is established using the `mysql2` host and other properties are the same as provided in the connection array. 

> And the next test as well.
`tenant database can be created on a foreign server by using the username and password from tenant config`

This test is same as above but it ensures that the tenant database connection can be established successfully from the tenantConfig username and password.

Reply:
If yes, then I realized that there's one thing we may need to resolve. I realized this a few days ago.

The `PermissionControlled...` DB manager uses `tenancy_db_username` and `tenancy_db_password` **as details for the user _that will be created_**.

But what if we wanted to specify details for the host connection, including:
- the hostname
- the username (of the user that will be used **for** creating the tenant's DB user)
- the password (also of the main user)

How could we do that?

Reply:
> Okay, so the tenancy_db_host overrides details in the template host connection, right?

Yes right. 

> But what if we wanted to specify details for the host connection, including:
the hostname
the username (of the user that will be used for creating the tenant's DB user)
the password (also of the main user)

I'm afraid I did not understand your point. Do you mean how to specify the hostname along with username and password?

Reply:
The review is resolved here https://github.com/archtechx/tenancy/pull/946 after discussing it internally.